trunk/SUAVE/Methods/Aerodynamics/Supersonic_Zero/Drag/compressibility_drag_total.py

```diff
@@ -215,18 +215,40 @@ def drag_div(Mc_ii,wing,k,cl,Sref_main):
 
         # Divergence mach number
         MDiv = np.array([[0.95]] * len(Mc_ii))
-        mcc = np.array([[0.93]] * len(Mc_ii))
+        mcc  = np.array([[0.93]] * len(Mc_ii))
 
     else:
         # Unpack wing
         t_c_w   = wing.thickness_to_chord
-        sweep_w = wing.sweeps.quarter_chord
+        if len(wing.Segments.keys())>0: # if wing has segments
+            symm           = wing.symmetric
+            semispan       = wing.spans.projected*0.5 * (2 - symm)
+            root_chord     = wing.chords.root
+            num_segments   = len(wing.Segments.keys())     
+            
+            weighted_sweep = 0
+            Sref           = 0
+            for i_segs in xrange(num_segments): 
+                if i_segs == num_segments-1:
+                    continue 
+                else:                    
+                    span_seg        = semispan*(wing.Segments[i_segs+1].percent_span_location - wing.Segments[i_segs].percent_span_location )
```

Comment:
This is a higher level SUAVE question. But this could be made as a sizing function instead. Then it only needs to be called once. If we call it in the analysis we are calling this code hundreds of times in a mission.

Reply:
So we discussed that the compressibility drag correlation is for the full wing and it would be incorrect to perform this on individual segments then summing the drag. I corrected it in the subsonic fidelity zero compressibility drag computation but forgot to correct the supersonic comp. drag computation. Issue resolved